src/commands/epgsql_cmd_connect.erl

```diff
@@ -47,22 +47,52 @@
 init(#{host := _, username := _} = Opts) ->
     #connect{opts = Opts}.
 
-execute(PgSock, #connect{opts = #{host := Host} = Opts, stage = connect} = State) ->
-    Timeout = maps:get(timeout, Opts, 5000),
-    Deadline = deadline(Timeout),
-    Port = maps:get(port, Opts, 5432),
+execute(PgSock, #connect{opts = #{username := Username} = Opts, stage = connect} = State) ->
     SockOpts = [{active, false}, {packet, raw}, binary, {nodelay, true}, {keepalive, true}],
-    case gen_tcp:connect(Host, Port, SockOpts, Timeout) of
-        {ok, Sock} ->
-            client_handshake(Sock, PgSock, State, Deadline);
+    PgSock1 = epgsql_sock:set_attr(connect_opts, Opts, PgSock),
```

Comment:
Opts includes `password`, which is a `fun()`. If you will store it in a permanent state it will crash during code upgrade. I think we should filter and store only necessary options there.